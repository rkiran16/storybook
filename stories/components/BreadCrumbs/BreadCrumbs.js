import { html } from "lit-html";

export const BreadCrumbs = ({overflow, disabled}) => {
   const data = [1,2,3];
   const disabledClass = disabled ? "disabled": "";
   const defaultHtml = () => html`
      <li class=${["breadcrumb-item", disabledClass].join(" ")}><a href='#'>Breadcrumb 1</a></li>
      <li class=${["breadcrumb-item", disabledClass].join(" ")}><a href='#'>Breadcrumb 2</a></li>
      <li class=${["breadcrumb-item", disabledClass, "active"].join(" ")}>Breadcrumb 3</li>
    `;

   const overflowHtml = () => html`
      <li class="breadcrumb-item d-flex">
        <div class="dropdown">
          <button class="btn btn-link dropdown-toggle" type="button" id="dropdownMenuButton" data-bs-toggle="dropdown" aria-expanded="false">
            ...
          </button>
          <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
            <a class="dropdown-item" href="#">Breadcrumb 1</a>
            <a class="dropdown-item" href="#">Breadcrumb 2</a>
            <a class="dropdown-item" href="#">Breadcrumb 3</a>
          </div>
        </div>
      </li>
      <li class="breadcrumb-item"><a href="#">Breadcrumb 4</a></li>
      <li class="breadcrumb-item active" aria-current="page">Breadcrumb 5</li>
   `;

   return html `
    <nav aria-label="breadcrumb">
      <ol class="breadcrumb">
        ${overflow ? overflowHtml() : defaultHtml()}
      </ol>
   </nav>
   `;
}