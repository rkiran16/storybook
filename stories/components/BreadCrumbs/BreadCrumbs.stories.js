import { BreadCrumbs } from "./BreadCrumbs";

export default {
   title: "Components/BreadCrumbs",
   parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62ab81201300101c6d1dcba7",
   },
   argTypes: {
     overflow: {
      name: "Show Overflow", 
      type: { name: 'boolean', required: false},
      description: "Show overflow Component",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
     },
     disabled: {
      name: "Disabled", 
      type: { name: 'boolean', required: false},
      description: "Disable Breadcrumb",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
     }
   }
}

const Template = (args) => BreadCrumbs(args);

export const Standard = Template.bind({});

export const OverflowBreadcrumb = Template.bind({});
OverflowBreadcrumb.args = {
   name: "Overflow Breadcrumb",
   overflow: true
}