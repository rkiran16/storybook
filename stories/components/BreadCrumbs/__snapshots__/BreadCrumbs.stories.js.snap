// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/BreadCrumbs Overflow Breadcrumb 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <nav aria-label=\\"breadcrumb\\">
      <ol class=\\"breadcrumb\\">
        ",
    "
      </ol>
   </nav>
   ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <li class=\\"breadcrumb-item d-flex\\">
        <div class=\\"dropdown\\">
          <button class=\\"btn btn-link dropdown-toggle\\" type=\\"button\\" id=\\"dropdownMenuButton\\" data-bs-toggle=\\"dropdown\\" aria-expanded=\\"false\\">
            ...
          </button>
          <div class=\\"dropdown-menu\\" aria-labelledby=\\"dropdownMenuButton\\">
            <a class=\\"dropdown-item\\" href=\\"#\\">Breadcrumb 1</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">Breadcrumb 2</a>
            <a class=\\"dropdown-item\\" href=\\"#\\">Breadcrumb 3</a>
          </div>
        </div>
      </li>
      <li class=\\"breadcrumb-item\\"><a href=\\"#\\">Breadcrumb 4</a></li>
      <li class=\\"breadcrumb-item active\\" aria-current=\\"page\\">Breadcrumb 5</li>
   ",
      ],
      "values": Array [],
    },
  ],
}
`;

exports[`Storyshots Components/BreadCrumbs Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <nav aria-label=\\"breadcrumb\\">
      <ol class=\\"breadcrumb\\">
        ",
    "
      </ol>
   </nav>
   ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <li class=",
        "><a href='#'>Breadcrumb 1</a></li>
      <li class=",
        "><a href='#'>Breadcrumb 2</a></li>
      <li class=",
        ">Breadcrumb 3</li>
    ",
      ],
      "values": Array [
        "breadcrumb-item ",
        "breadcrumb-item ",
        "breadcrumb-item  active",
      ],
    },
  ],
}
`;
